numpy/polynomial/_polybase.py

```diff
@@ -284,9 +284,41 @@ def __repr__(self):
         return f"{name}({coef}, domain={domain}, window={window})"
 
     def __str__(self):
-        coef = str(self.coef)
-        name = self.nickname
-        return f"{name}({coef})"
+        # Get configuration for line breaks
+        linewidth = np.get_printoptions().get('linewidth', 75)
+        if linewidth < 1:
+            linewidth = 1
+        out = ""
+        if self.coef[0] < 0:
+            out += "-"
+        out = f"{self.coef[0]} "
+        cur_line = len(out) // linewidth    # For handling linebreaks
+        for i, coef in enumerate(self.coef[1:]):
+            next_term = ""
+            power = str(i + 1)
+            # Polynomial coefficient
+            if coef >= 0:
```

Comment:
do we need to print 0 coef terms ? Is this to keep it consistent with latex representation ? my personal pref would be if its not blurred to omit it, since it may not be very readable in some cases for eg. sparse polynomials (as pointed out by @philer in https://github.com/numpy/numpy/issues/8893 )

Reply:
No, the 0-coef terms don't need to be printed. I erred on the side of caution and decided to keep them in so as not to be "hiding" information from the user (after all, the 0's are part of the polynomial's `coef`).

Note that the `_repr_latex` keeps the zero terms, but "grays them out". This is a nice solution but I don't know if there is a comparable approach that could work for terminal output given that the support for rich output is so limited and platform/terminal dependent.